coalib/output/Logging.py

```diff
@@ -1,28 +1,80 @@
 from datetime import datetime
-import json
 import io
+import json
 import logging
 import logging.config
 
+from coalib.output.printers.LOG_LEVEL import LOG_LEVEL
+
 
-def configure_logging():
+def configure_logging(new_log_level=logging.INFO, incremental=False,
+                      stdout=False, color=True):
     """
     Configures the logging with hard coded dictionary.
+
+    If the incremental param is ``True``, all the log-handlers configuration
+    other than level will be ignored (including class).
+    Thus, use it only when changing the log_level.
+    When changing log_level, ``json-handler``, ``console-handler``,
+    and ``file-handler`` should all exist.
+    json-handler is created as a null handler to avoid this error.
+
+    :param new_log_level:
+        The new log level to set for json and console handlers.
+    :param incremental:
+        ``False`` if setting up new loggers, ``True`` if only changing
+        the log level.
+    :param stdout:
+        If ``True`` the logging is done on stdout, else its done stderr.
+    :param color:
+        If ``True`` the logging uses ``color`` formatter, else it uses
+        the ``no-color`` formatter.
+    :return:
+        The new_log_level is returned.
+
+    >>> from coalib.output.Logging import configure_logging
+    >>> import logging
+    >>> import sys
+    >>> configure_logging(stdout=True, color=False)
+    20
+    >>> logging.info("This world doesn’t belong to them")
+    [INFO]... This world doesn’t belong to them
+
+    We will now change the level to ERROR.
+
+    >>> configure_logging(logging.ERROR, incremental=True)
+    40
+    >>> logging.info("it belongs to us.")
+    >>> logging.error("These violent delights have violent ends.")
+    [ERROR]... These violent delights have violent ends.
     """
     import sys
 
     logging.config.dictConfig({
         'version': 1,
+        'incremental': incremental,
         'handlers': {
             'console-handler': {
                 'class': 'logging.StreamHandler',
-                'formatter': 'color',
-                'stream': sys.stderr
+                'formatter': 'color' if color else 'no-color',
+                'stream': sys.stdout if stdout else sys.stderr,
```

Comment:
Going to make one commit for color and stream, logical change -> add more control to logging?